uvacw/osd2f
# Feature/mvp visualisation

# Implemented V1 filtering interface to give respondents control over their donation

## Closes #7 
____
Implemented an inspection & redaction interface. This includes:
* Option to curate a donation by clicking the "Inspect & edit your donation" button
* Short explainer tab 
* per-file table of contents adaptive to underlying data. 
* Search-functionality to find records per file
* Select & Remove functionality
* Counter per file of number of removed entries
* pagination of data
* pagination of files
* Consent modal popup with "consent" and non-consent ("I'm sorry") option

## Assumptions

This PR introduces the [Vue](https://vuejs.org/) front-end framework to create a client-side interface for data filtering. It leverages [bootstrap-vue](https://bootstrap-vue.org/) to keep the formatting consistent. The webpack configuration is kept to a minimum. This may mean that we'll need to add babel/polyfills later if we run into old browsers that we want to support. 

I'm hoping Vue will give us a relativily easy cross-filtering mechanism to provide visualizations that can also be used to search through the content. This is currently out of scope though. 

## Usage / Minimal Example

Instructions on how to verify changes this PR by running code

1. Start the server

```bash
osd2f -m Development -vvv
```
2. Select a (zip) file or drag in a folder/file
3. Click around the interface, try searching etcetera
4. When submitting, check whether the received number of submissions matches the amount áfter redacting the data

## Preview

![osd2f_submission_mvp](https://user-images.githubusercontent.com/1457643/108919365-2882c300-7633-11eb-89b3-627db55f16ab.gif)


## Checklist
- [ ] Added tests if appropriate (and it should always be)
- [X] Created new issues when required
